compiler/luci/pass/src/FuseMulWithFullyConnectedPass.test.cpp

```diff
@@ -0,0 +1,254 @@
+
+  auto fc = dynamic_cast<luci::CircleFullyConnected *>(g.output()->from());
+  EXPECT_NE(nullptr, fc);
+
+  auto weights = loco::must_cast<luci::CircleConst *>(g.fc()->weights());
+  auto weights_n = weights->dim(0).value();
+  auto weights_m = weights->dim(1).value();
+  uint32_t offset = 0;
+  for (uint32_t i = 0; i < weights_n; i++)
+  {
+    for (uint32_t j = 0; j < weights_m; j++)
+    {
+      offset = i * weights_m + j;
+      EXPECT_EQ(i * offset, weights->at<loco::DataType::FLOAT32>(offset));
+    }
+  }
+
+  auto bias = loco::must_cast<luci::CircleConst *>(g.fc()->bias());
+  for (uint32_t i = 0; i < bias->size<loco::DataType::FLOAT32>(); i++)
+  {
+    EXPECT_EQ(i * i, bias->at<loco::DataType::FLOAT32>(i));
+  }
+}
+
+TEST_F(FuseMulWithFullyConnectedPassTest, fc_mul_scalar)
+{
+  g.init(luci::FusedActFunc::NONE, true /* is_mul_scalar */, true /* use_bias */);
+
+  EXPECT_EQ(true, pass.run(g.g()));
+
+  auto fc = dynamic_cast<luci::CircleFullyConnected *>(g.output()->from());
+  EXPECT_NE(nullptr, fc);
+
+  auto weights = loco::must_cast<luci::CircleConst *>(g.fc()->weights());
+  auto weights_n = weights->dim(0).value();
+  auto weights_m = weights->dim(1).value();
+  uint32_t offset = 0;
+  for (uint32_t i = 0; i < weights_n; i++)
+  {
+    for (uint32_t j = 0; j < weights_m; j++)
+    {
+      offset = i * weights_m + j;
+      EXPECT_EQ(MUL_VAL * offset, weights->at<loco::DataType::FLOAT32>(offset));
+    }
+  }
+
+  auto bias = loco::must_cast<luci::CircleConst *>(g.fc()->bias());
+  for (uint32_t i = 0; i < bias->size<loco::DataType::FLOAT32>(); i++)
+  {
+    EXPECT_EQ(MUL_VAL * i, bias->at<loco::DataType::FLOAT32>(i));
+  }
+}
+
+TEST_F(FuseMulWithFullyConnectedPassTest, fc_no_bias)
+{
+  g.init(luci::FusedActFunc::NONE, false /* is_mul_scalar */, false /* use_bias */);
+
+  EXPECT_EQ(true, pass.run(g.g()));
+
+  auto fc = dynamic_cast<luci::CircleFullyConnected *>(g.output()->from());
+  EXPECT_NE(nullptr, fc);
+  auto no_bias = dynamic_cast<luci::CircleOutputExclude *>(fc->bias());
+  ASSERT_NE(nullptr, no_bias);
+
+  auto weights = loco::must_cast<luci::CircleConst *>(g.fc()->weights());
+  auto weights_n = weights->dim(0).value();
+  auto weights_m = weights->dim(1).value();
+  uint32_t offset = 0;
+  for (uint32_t i = 0; i < weights_n; i++)
+  {
+    for (uint32_t j = 0; j < weights_m; j++)
+    {
+      offset = i * weights_m + j;
+      EXPECT_EQ(i * offset, weights->at<loco::DataType::FLOAT32>(offset));
+    }
+  }
+}
+
+TEST_F(FuseMulWithFullyConnectedPassTest, bias_feature_map_NEG)
+{
+  g.init(luci::FusedActFunc::NONE, false /* is_mul_scalar */, true /* use_bias */);
+
+  // Bias cannot be fused as it's passed as feature map.
+  g.to_fm_bias();
+
+  EXPECT_EQ(false, pass.run(g.g()));
+}
+
+TEST_F(FuseMulWithFullyConnectedPassTest, fc_with_activation_NEG)
+{
+  g.init(luci::FusedActFunc::RELU, false /* is_mul_scalar */, true /* use_bias */);
+
+  EXPECT_EQ(false, pass.run(g.g()));
+}
```

Comment:
please make it as count(positive) <= count(negative)

Reply:
They are matching now.